Marlin/Marlin_main.cpp

```diff
@@ -3298,6 +3410,45 @@ inline void gcode_G28() {
         } //xProbe
       } //yProbe
 
+      #if ENABLED(PROBE_FAIL_PANIC)
+        if(reprobe_attempts == NUM_ATTEMPTS && probePointCounter == -1)
+        {
+          if(!IS_SD_PRINTING)
+            probe_fail = true;
+            disable_all_heaters();
```

Comment:
This indentation implies that `disable_all_heaters()` is part of the `if(!IS_SD_PRINTING)` block, but only `probe_fail = true;` is executed when `if(!IS_SD_PRINTING)` here.